lib/player/stream_video_source.js

```diff
@@ -1110,7 +1110,13 @@ shaka.player.StreamVideoSource.prototype.startStreams_ = function(
   // Set the video's current time before starting the streams so that the
   // streams begin buffering at the stream start time.
   shaka.log.info('Starting each stream from', streamStartTime);
-  this.video.currentTime = streamStartTime;
+
+  if (this.video.readyState >= HTMLMediaElement.HAVE_METADATA) {
+    this.video.currentTime = streamStartTime;
+  } else {
+    this.video.addEventListener('loadedmetadata',
```

Comment:
We have a class called EventManager which is used to avoid memory leaks that can be caused by event listeners.  If 'loadedmetadata' never fires, this StreamVideoSource could be retained in memory, or worse, called on a subsequent playback after this StreamVideoSource has been destroyed.

If you use this.eventManager.listen and unlisten, we can avoid that.